src/main/java/openmodularturrets/blocks/misc/LeverBlock.java

```diff
@@ -72,65 +74,79 @@ public void onBlockPlacedBy(World par1World, int par2, int par3, int par4, Entit
+        int xReal = x;
+        int zReal = z;
+        switch (turretDirection) {
+            case NORTH:
+                z--; // Assuming North is negative Z in your world
+                break;
+            case SOUTH:
+                z++; // And South is positive Z
+                break;
+            case EAST:
+                x++; // East is positive X
+                break;
+            case WEST:
+                x--; // West is negative X
+                break;
+            default:
+                throw new IllegalStateException(
+                        "The metadata led to an unknown direction. This shouldn't happen, see?");
         }
 
-        if ((par1World.getBlockMetadata(par2, par3, par4) * 90) == 90
-                && par1World.getTileEntity(par2 - 1, par3, par4) instanceof TurretBaseTierOneTileEntity) {
-            base = (TurretBaseTierOneTileEntity) par1World.getTileEntity(par2 - 1, par3, par4);
-            if (base != null) {
-                lever.isTurning = true;
-                if (lever.rotation == 0F) {
-                    par1World.playSoundEffect(par2, par3, par4, "openmodularturrets:windup", 1.0F, 1.0F);
-                    base.receiveEnergy(ForgeDirection.UNKNOWN, 50, false);
-                }
-            }
+        if (!(world.getTileEntity(x, y, z) instanceof TurretBaseTierOneTileEntity)) {
+            this.breakBlock(world, xReal, y, zReal, this, world.getBlockMetadata(xReal, y, zReal));
         }
 
-        if ((par1World.getBlockMetadata(par2, par3, par4) * 90) == 180
-                && par1World.getTileEntity(par2, par3, par4 - 1) instanceof TurretBaseTierOneTileEntity) {
-            base = (TurretBaseTierOneTileEntity) par1World.getTileEntity(par2, par3, par4 - 1);
-            if (base != null) {
-                lever.isTurning = true;
-                if (lever.rotation == 0F) {
-                    par1World.playSoundEffect(par2, par3, par4, "openmodularturrets:windup", 1.0F, 1.0F);
-                    base.receiveEnergy(ForgeDirection.UNKNOWN, 50, false);
-                }
-            }
+        return (TurretBaseTierOneTileEntity) world.getTileEntity(x, y, z);
+    }
+
+    private static ForgeDirection decipherMetadata(int metadata) {
+        // Your metadata deciphering logic goes here
+        // For now, let's just pretend it's a simple mapping
+        switch (metadata) {
+            case 0:
+                return SOUTH;
+            case 1:
+                return WEST;
+            case 2:
+                return NORTH;
+            case 3:
+                return EAST;
+            default:
+                throw new IllegalArgumentException(
+                        "Bad metadata, kid. It's like reading a book with half the pages torn out.");
         }
+    }
+
+    @Override
+    public boolean onBlockActivated(World par1World, int par2, int par3, int par4, EntityPlayer par5EntityPlayer,
+            int par6, float par7, float par8, float par9) {
```

Comment:
Please name things correctly.
```suggestion
    public boolean onBlockActivated(World worldIn, int x, int y, int z, EntityPlayer player, int side, float subX, float subY, float subZ) {
```